uast/uast_test.go

```diff
@@ -30,3 +30,118 @@ func TestNodeTokens(t *testing.T) {
+
+	n := &Node{InternalType: "empty",
+		Children: []*Node{
+			{InternalType: "id", Token: "id3"},
+			{InternalType: "op_prefix", Roles: []Role{Prefix}, Token: "Prefix+", Children: []*Node{
+				{InternalType: "left", Token: "tok_pre_left"},
+				{InternalType: "right", Token: "tok_pre_right"},
+			}}}}
+	result := Tokens(n)
+	expected := []string{"id3", "Prefix+", "tok_pre_left", "tok_pre_right"}
+	require.Equal(expected, result)
+}
+
+func TestPrefixTokensPlain(t *testing.T) {
+	require := require.New(t)
+
+	n := &Node{InternalType: "empty",
+		Children: []*Node{
+			{InternalType: "id", Token: "id3"},
+			{InternalType: "op_prefix", Roles: []Role{Prefix}, Token: "Prefix+"},
+			{InternalType: "left", Token: "tok_pre_left"},
+			{InternalType: "right", Token: "tok_pre_right"},
+			}}
+	result := Tokens(n)
+	expected := []string{"id3", "Prefix+", "tok_pre_left", "tok_pre_right"}
+	require.Equal(expected, result)
+}
+
+func TestInfixTokens(t *testing.T) {
+	require := require.New(t)
+	n := &Node{InternalType: "empty",
+		Children: []*Node{
+			{InternalType: "id", Token: "id1"},
+			{InternalType: "op_infix", Roles: []Role{Infix}, Token: "Infix+", Children: []*Node{
+				{InternalType: "left", Token: "tok_in_left"},
+				{InternalType: "right", Token: "tok_in_right"},
+			}}}}
+	result := Tokens(n)
+	expected := []string{"id1", "tok_in_left", "Infix+", "tok_in_right"}
+	require.Equal(expected, result)
+}
+
+func TestInfixTokensPlain(t *testing.T) {
+	require := require.New(t)
+	n := &Node{InternalType: "empty",
+		Children: []*Node{
+			{InternalType: "id", Token: "id1"},
+			{InternalType: "left", Token: "tok_in_left"},
+			{InternalType: "op_infix", Roles: []Role{Infix}, Token: "Infix+"},
+			{InternalType: "right", Token: "tok_in_right"},
+			}}
+	result := Tokens(n)
+	expected := []string{"id1", "tok_in_left", "Infix+", "tok_in_right"}
+	require.Equal(expected, result)
+}
+
+func TestPostfixTokens(t *testing.T) {
+	require := require.New(t)
+	n := &Node{InternalType: "empty",
+		Children: []*Node{
+			{InternalType: "id", Token: "id2"},
+			{InternalType: "op_postfix", Roles: []Role{Postfix}, Token: "Postfix+", Children: []*Node{
+				{InternalType: "left", Token: "tok_post_left"},
+				{InternalType: "right", Token: "tok_post_right"},
+			}}}}
+	result := Tokens(n)
+	expected := []string{"id2", "tok_post_left", "tok_post_right", "Postfix+"}
+	require.Equal(expected, result)
+}
+
+func TestPostfixTokensPlain(t *testing.T) {
+	require := require.New(t)
+	n := &Node{InternalType: "empty",
+		Children: []*Node{
+			{InternalType: "id", Token: "id2"},
+			{InternalType: "left", Token: "tok_post_left"},
+			{InternalType: "right", Token: "tok_post_right"},
+			{InternalType: "op_postfix", Roles: []Role{Postfix}, Token: "Postfix+"},
+			}}
+	result := Tokens(n)
+	expected := []string{"id2", "tok_post_left", "tok_post_right", "Postfix+"}
+	require.Equal(expected, result)
+}
+
+// Test for mixed order roles
+func TestOrderTokens(t *testing.T) {
```

Comment:
:+1:  to all. I should really make the mouse give me an electrical discharge every time I write "childs". I'll add the test with further tokens as a separate case if you don't mind because I like the unittests to be as atomic as possible.